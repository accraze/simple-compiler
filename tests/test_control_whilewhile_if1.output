Name= j  Value= 0
Name= i  Value= 0
Name= minusfour  Value= 0
Name= k  Value= 0
Error: invalid value for stmt_type (0).
