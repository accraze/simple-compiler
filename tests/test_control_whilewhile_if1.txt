
j, i, minusfour, k ;

{
i = 4;
j = 3;
minusfour = 0-4;

WHILE i > 0 {
	j = 3;	
	IF j > 1 {	
		WHILE j > 0 {
			k = i*j;
			print k;
			j = j-1;
		}
	}
	IF j > 0 {
		WHILE j > 0 {
			k = i*j;
			print k;
			j = j-1;
		}
	}
	i = i-1;
}

WHILE i > minusfour {
	print i;
	i = i - 1;
}

i = i+1;
j = j+1;
print i;
print j;

}
