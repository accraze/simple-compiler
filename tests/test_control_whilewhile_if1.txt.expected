12
8
4
9
6
3
6
4
2
3
2
1
0
-1
-2
-3
-3
1

